src/bytes_mut.rs

```diff
@@ -1035,6 +1034,16 @@ impl BytesMut {
             slice::from_raw_parts_mut(ptr.cast(), len)
         }
     }
+
+    /// Rebuild a vec from `offset` with the existing pointer, length, and capacity.
+    #[inline]
+    unsafe fn rebuild_vec(&self, offset: usize) -> Vec<u8> {
```

Comment:
Can you document a bit about why this is unsafe? That is, explain that this transfers ownership of the allocation to the vector, but that it doesn't remove the pointers from the `BytesMut` that it is called on. (And of course, the offset must be correct.)

I know that it didn't explain this before your change, but by moving it to `BytesMut`, you're introducing the connotation that it could modify the `BytesMut` to no longer own the allocation. That connotation was not there when it was a standalone method.

Reply:
Thanks for pointing that out @Darksonn! I added some information in f62a4d03b9a714bb90d563832158a3048689e841. Let me know if you prefer something different.

Also, the source of `offset` is always the offset from `self.get_vec_pos`, so we could probably remove that requirement by fetching it in this method. I'll attempt that separately though!